doc/address.md

```diff
@@ -0,0 +1,154 @@
+# Dogecoin Addresses
+
+A Dogecoin Address is the Public Key of an assymetric key-pair, and provides
+a designation on the network to hold Dogecoins on the ledger. By holding the
+Private Key for an Address on the Ledger you have the ability to sign 
+Transactions to move (spend/send) the Dogecoins attributed to the Address.
+
+Dogecoin Addresses can be created in a number of ways, each with benefits
+depending on the objective of the Address owner(s). 
+
+## Simple Addresses
+
+These are the most common form of Dogecoin Address, a single Private Key with
+a single Public Key (Address). Most often these are what an exchange would use
+to retain full custody of a wallet on behalf of their users, or for a simple
+desktop / mobile wallet they provide a basic approach to Address management.
+
+#### Essential APIs
+
+The High level 'essential' API provided by Libdogecoin for working with Simple 
+Addresses are found in `include/dogecoin/address.h` and are:
+
+**generatePrivPubKeypair:**
+
+`generatePrivPubKeypair(char* wif_privkey, char* p2pkh_pubkey, bool is_testnet)`
+
+This function will populate provided private/public char* variables with a freshly
+generated private/public key pair, specifically for the network specified by the
+third parameter. 
+
+
+_C usage:_
+```C
+#include <dogecoin/address.h>
+int keyLen = 100; 
+char privKey[keyLen]; 
+char pubKey[keyLen]; 
+generatePrivPubKeypair(privKey, pubKey, false);
+```
+
+_Python usage:_
+```py
+import libdogecoin
+
+# generatePrivPubKeypair returns a two-touple of pub/priv key strings
+keys = libdogecoin.generatePrivPubKeypair()
```

Comment:
needs to include a dogecoin chain indicator, either 0 for main or 1 for testnet. My current format, feel free to change:
keys = libdogecoin.generatePrivPubKeypair(chain_code)

Reply:
Oh ok, I thought in python it would be implemented with a default arg to false ?

Reply:
That's exactly right IMO,  0 for main 1 for testnet.  However we have it taking in a 'blank' / 'allocated' buffer for privkey and pubkey vs doing a 'return'.  (it's not great to return more than one thing. It can be done with a lot of work but... it's best if the allocated privkey and pubkey buffers just get sent in as a parameter and once the call is done, they're populated. Byref basically.)

Reply:
sure i can implement that, default main would make sense

Reply:
yeah its still byref, the chain indicator is just for the boolean is_testnet, but yeah i'll change the py function so that it has a default arg since most users probably will just be using main net

Reply:
Ok so Tim informs me that this would return a tuple of privkey, pubkey - and that's fine! Sorry, my shame is great and I need to python moar. :)

Reply:
I just wrote these examples to start the conversation rolling :) This is all really great stuff everyone!  I think if we can get through this address.md doc together and spec out what the high level functions should look like (including code examples) then it's a lot easier to validate we're building something useful in the end?